terraform/diff.go

```diff
@@ -279,6 +311,10 @@ func (d *ModuleDiff) String() string {
 	return buf.String()
 }
 
+func (d *ModuleDiff) GoString() string {
```

Comment:
I don't think we need these GoString methods.
It's only replacing the `&` in the standard GoString format with a `*`, but causes vet to fail if we need to add a sync.* field at some point. I think it would be better to be consistent with the other structures that will have the `&` prepended with the default format. (I've already removed/fixed the ones that caused failures, but can clean them all up at some point)